internal/api/websocket.go

```diff
@@ -239,19 +247,18 @@ func newWebSocketProxy(connection webSocketConnection) (*webSocketProxy, error)
 // waitForExit waits for an exit of either the runner (when the command terminates) or the client closing the WebSocket
 // and handles WebSocket exit messages.
 func (wp *webSocketProxy) waitForExit(exit <-chan runner.ExitInfo, cancelExecution context.CancelFunc) {
-	cancelInputLoop := wp.Stdin.startReadInputLoop()
+	wp.Stdin.startReadInputLoop()
+
 	var exitInfo runner.ExitInfo
 	select {
-	case exitInfo = <-exit:
-		cancelInputLoop()
-		wp.cancelWebSocketWrite()
-		log.Info("Execution returned")
-	case <-wp.userExit:
-		cancelInputLoop()
-		wp.cancelWebSocketWrite()
-		cancelExecution()
+	case <-wp.webSocketCtx.Done():
 		log.Info("Client closed the connection")
+		cancelExecution()
+		<-time.After(ProxyShutdownGracePeriod)
```

Comment:
You probably tried: We cannot use a channel to wait until the SIGQUIT was sent, right? Waiting a second here is fine for me as well ;)

Reply:
Yes, I tried. But I always got lost in too many changes.
Now I have added a better solution.

Reply:
I like that, it's much cleaner ;)